server/src/main/java/org/apache/druid/server/scheduling/HiLoQueryLaningStrategy.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.scheduling;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Ints;
+import it.unimi.dsi.fastutil.objects.Object2IntArrayMap;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import org.apache.druid.client.SegmentServerSelector;
+import org.apache.druid.query.Query;
+import org.apache.druid.query.QueryContexts;
+import org.apache.druid.query.QueryPlus;
+import org.apache.druid.server.QueryLaningStrategy;
+
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Query laning strategy which associates all {@link Query} with priority lower than 0 into a 'low' lane
+ */
+public class HiLoQueryLaningStrategy implements QueryLaningStrategy
+{
+  public static final String LOW = "low";
+
+  @JsonProperty
+  private final int maxLowPercent;
```

Comment:
Does it make sense to add `maxCapacity` as well?

Reply:
hmm, this is the maximum number of concurrent queries for this lane.

`druid.query.scheduler.numThreads` or `druid.server.http.numThreads` define the maximum number of concurrent running queries, but it's not lane specific.

Reply:
I mean, sometimes you may want to increase the total capacity, but keep it as it is for the low lane. In that case, `maxCapacity` would be more convenient because you don't have to change the configuration.

Reply:
Ah, so do you mean the lane limit should be the min or (or max?) of `maxCapacity` and `maxLowPercent` of `totalCapacity`?

Reply:
We could do that. Or we can allow either `maxLowPercent` or `maxCapacity` to be set.

Reply:
Would it be ok to do this addition in a follow-up PR? Originally this setting was `maxLowThreads`, but i switched to being percent based to be more versatile. However, it does seem maybe useful to allow one or the other to be set.

Reply:
Sounds good to me.